Tests/SotoTests/Services/Lambda/LambdaTests.swift

```diff
@@ -29,15 +113,42 @@ class LambdaTests: XCTestCase {
             print("Connecting to AWS")
         }
 
-        Self.client = AWSClient(credentialProvider: TestEnvironment.credentialProvider, middlewares: TestEnvironment.middlewares, httpClientProvider: .createNew)
+        Self.client = AWSClient(
+            credentialProvider: TestEnvironment.credentialProvider,
+            middlewares: TestEnvironment.middlewares,
+            httpClientProvider: .createNew
+        )
         Self.lambda = Lambda(
             client: LambdaTests.client,
             region: .euwest1,
             endpoint: TestEnvironment.getEndPoint(environment: "LOCALSTACK_ENDPOINT")
         )
+        Self.iam = IAM(
+            client: Self.client,
+            endpoint: TestEnvironment.getEndPoint(environment: "LOCALSTACK_ENDPOINT")
+        )
+
+        // create a lambda function to test invoke()
+        do {
+            let response = try Self.createIAMRole().wait()
+            // IAM needs some time after Role creation, before the role can be attached to a Lambda function
+            // https://stackoverflow.com/a/37438525/663360
+            print("Sleeping 15 secs, waiting for IAM Role to be created")
+            sleep(15)
+            _ = try Self.createLambdaFunction(roleArn: response.role.arn).wait()
+        } catch {
+            XCTFail("Can not create prerequisites resources in the cloud, before to execute the tests: \(error)")
+        }
```

Comment:
Can we use `XCTAssertNoThrow` here instead. If you chain the two commands together you don't need to worry about accessing the response from the createIAMRole call.
```
        let response  = Self.createIAMRole().flatMap { response -> EventLoopFuture<Lambda.FunctionConfiguration> in
            let eventLoop = Self.client.eventLoopGroup.next()
            print("Sleeping 15 secs, waiting for IAM Role to be created")
            let scheduled = eventLoop.flatScheduleTask(in: .seconds(15)) {
                Self.createLambdaFunction(roleArn: response.role.arn)
            }
            return scheduled.futureResult
        }
        XCTAssertNoThrow(try response.wait())
```

Reply:
nice implementation, also better aligned with other tests